src/main/scala/Switcher.scala

```diff
@@ -0,0 +1,147 @@
+package testchipip
+
+import chisel3._
+import chisel3.util._
+import freechips.rocketchip.config.Parameters
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+
+class TLSplitter(n: Int, params: TLBundleParameters) extends Module {
+  val io = IO(new Bundle {
+    val in = Flipped(new TLBundle(params))
+    val out = Vec(n, new TLBundle(params))
+    val sel = Input(UInt(log2Ceil(n).W))
+  })
+
+  io.out.zipWithIndex.foreach { case (out, i) =>
+    val selected = io.sel === i.U
+    out.a.valid := io.in.a.valid && selected
+    out.a.bits := io.in.a.bits
+    out.d.ready := io.in.d.ready && selected
+
+    out.b.ready := false.B
```

Comment:
Can we have an assertion on io.in.b.valid (and the other unused channels)?

Reply:
It will just freeze up if it actually tries to send, won't it? Since the ready is false.

Reply:
Yeah I just want the assertion for simulation. Its easy to debug an assertion than a timeout.